yarn-project/bot/src/factory.ts

```
@@ -54,7 +56,14 @@ export class BotFactory {
       return account.register();
     } else {
       this.log.info(`Initializing account at ${account.getAddress().toString()}`);
-      return account.waitSetup({ timeout: this.config.txMinedWaitSeconds });
+      const sentTx = account.deploy();
+      if (this.config.flushSetupTransactions && this.node) {
```

I'd throw if flush is true but node is not set